lib/linguist/heuristics.rb

```diff
@@ -98,9 +101,13 @@ def matches?(filename, candidates)
     end
 
     # Internal: Perform the heuristic
-    def call(data)
+    def call(data, filename)
       matched = @rules.find do |rule|
-        rule['pattern'].match(data)
+        unless rule['filename_pattern'].nil?
```

Comment:
Eeeeew 🤢 This feels wrong to me. How about flipping it to if / else:

```suggestion
        if rule['filename_pattern']
```

Reply:
Sorry, I must've been thinking backwards. 😂 Again.

Fixed.